tests/tests.c

```diff
@@ -2207,11 +2209,17 @@ bool filter_error(bool pass, uint16_t err) {
 }
 
 uint32_t getrand(uint32_t upper_bound) {
-#if !defined(__WIN__)
+#if !defined(__WIN__) || defined(__EMBARCADEROC__)
 # if defined(__MACOS__) || defined(__BSD__)
+    if (upper_bound < 1)
```

Comment:
upper_bound could only be 0 if it were < 1? are you seeing zeroes?

Reply:
The arc4random page says to never ever pass a 0 so I'm just ensuring we never do.  As an aside, that function is available in newer versions of Linux and OpenBSD - should we look into using it when it's available?  I guess
Random() is fine here.

Reply:
But it also says that it returns a value less than upper bound, and uint32_t can't hold decimal places... so shouldn't it be a 2?

Reply:
And yes, we should use it wherever it's available... I'll check my lunixes. I think it's already enabled for FreeBSD

Reply:
This should be a 2, not a 1, if it returns "less than upper_bound"

Reply:
Not gonna let this one go either. You said that you should never pass a zero, and the man page says it returns less than upper_bound, which means if you pass 1 it can only return 0. At least with 2 it could be 0 or 1